ODIN_II/SRC/ast_elaborate.cpp

```diff
@@ -2216,36 +2216,43 @@ void update_instance_parameter_table(ast_node_t* instance, STRING_CACHE* instanc
-                /* update this parameter and remove all other overrides with this name */
-                free_whole_tree(((ast_node_t*)instance_param_table_sc->data[sc_spot]));
-                instance_param_table_sc->data[sc_spot] = (void*)parameter_override_list->children[j]->children[5];
-                parameter_override_list->children[j]->children[5] = NULL;
-
-                for (int k = 0; k < j; k++) {
-                    if (!strcmp(parameter_override_list->children[k]->children[0]->types.identifier, param_id)) {
-                        /**
+                ast_node_t* temp = (ast_node_t*)instance_param_table_sc->data[sc_spot];
+                if (temp->types.variable.is_parameter) {
+                    /* update this parameter and remove all other overrides with this name */
+                    free_whole_tree(((ast_node_t*)instance_param_table_sc->data[sc_spot]));
+                    instance_param_table_sc->data[sc_spot] = (void*)parameter_override_list->children[j]->children[5];
+                    parameter_override_list->children[j]->children[5] = NULL;
+
+                    for (int k = 0; k < j; k++) {
+                        if (!strcmp(parameter_override_list->children[k]->children[0]->types.identifier, param_id)) {
+                            /**
                          * TODO: is this needed?
                          * ast_node_t* temp = parameter_override_list->children[k]; 
                          * */
-                        remove_child_from_node_at_index(parameter_override_list, k);
-                        //temp = NULL;
-                        j--;
-                        k--;
+                            remove_child_from_node_at_index(parameter_override_list, k);
+                            //temp = NULL;
+                            j--;
+                            k--;
+                        }
                     }
-                }
-                /**
+                    /**
                  * TODO: is this needed?
                  * ast_node_t* temp = parameter_override_list->children[j];
                  * */
-                remove_child_from_node_at_index(parameter_override_list, j);
-                //temp = NULL;
+                    remove_child_from_node_at_index(parameter_override_list, j);
+                    //temp = NULL;
+                } else if (temp->types.variable.is_localparam) {
+                    warning_message(AST, parameter_override_list->loc,
+                                    "Defparam cannot override localparameter: %s\n",
+                                    parameter_override_list->children[j]->children[0]->types.identifier);
+                }
             }
         }
         // check if there are still overrides in the list that haven't been deleted
 
-        if (parameter_override_list->num_children > 0) {
-            oassert(false); // not reachable???
-        }
+        //if (parameter_override_list->num_children > 0) {
```

Comment:
make sure you keep this, guard since we should have covered everything otherwise there will be a memory leak